src/server_manager/web_app/ui_components/outline-server-view.js

```diff
@@ -902,41 +921,43 @@ export class ServerView extends DirMixin(PolymerElement) {
     return item.id !== MY_CONNECTION_USER_ID;
   }
 
+  _computeColumnDirection(columnName, accessKeySortBy, accessKeySortDirection) {
+    if (columnName === accessKeySortBy) {
+      return accessKeySortDirection;
+    }
+    return 0;
+  }
+
   _setSortByOrToggleDirection(e) {
     const sortBy = e.target.dataset.sortBy;
     if (this.accessKeySortBy !== sortBy) {
       this.accessKeySortBy = sortBy;
-      this.accessKeySortDirection = 1;  // Sort ascending initially.
-      return;
+      this.accessKeySortDirection = sortBy == 'usage' ? -1 : 1;
+    } else {
+      this.accessKeySortDirection = this.accessKeySortDirection * -1
     }
-    this.accessKeySortDirection *= -1;  // Toggle sort direction.
   }
 
   _sortAccessKeys(accessKeySortBy, accessKeySortDirection) {
     if (accessKeySortBy === 'usage') {
       return function(a, b) {
-        return (a.transferredBytes - b.transferredBytes) * this.accessKeySortDirection;
+        return (a.transferredBytes - b.transferredBytes) * accessKeySortDirection;
       }.bind(this);
     }
     // Default to sorting by name.
     return function(a, b) {
-      const nameA = a.name.toUpperCase() || a.placeholderName.toUpperCase();
-      const nameB = b.name.toUpperCase() || b.placeholderName.toUpperCase();
-      if (nameA < nameB) {
-        return -1 * this.accessKeySortDirection;
-      }
-      if (nameA > nameB) {
-        return 1 * this.accessKeySortDirection;
+      if (a.name && b.name) {
+        return compare(a.name.toUpperCase(), b.name.toUpperCase()) * accessKeySortDirection;
+      } else if (a.name) {
+        return -1;
+      } else if (b.name) {
+        return 1
+      } else {
+        return 0;
       }
-      return 0;
     }.bind(this);
```

Comment:
nit: it doesn't seem like we need to bind `this` anymore.

Reply:
Done